docs/proposals/console/devfile-import.md

```diff
@@ -0,0 +1,65 @@
+  version: 1.0.0
+  attributes:
+    alpha.build-dockerfile: https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/build/Dockerfile # can also be a path relative to the context
+components:
+- name: myapplication
+  attributes:
+    tool: console-import # key:value pair used to filter container type component that only the Console Devfile Import is interested in
+  container:
+    image: buildContextImageOutput:latest # this is the image which will be used by the Console's buildConfig output
+    endpoints:
+    - name: http-3000 # define endpoints in devfile.yaml, rather than hardcoding a default
+      targetPort: 3000
+    env:
+      - name: FOO # set container env through the devfile.yaml for the container
+        value: "bar"
+```
+
+Console's devfile import POC is using a `BuildConfig` to build the Dockerfile from the dockerfile path location, currently found from the context dir. The POC PR `BuildConfig` outputs it to an `ImageStreamTag`. However, the POC PR assumes the `ImageStream` and `ImageStreamTag` are all the same name as the application name. This is tightly coupled and dependant on the information entered in the Console Devfile Import form and thus does not allow us to mention a custom image name in the devfile.
+
+If we want to decouple `ImageStream` and `ImageStreamTag` from the application name, so that a custom image name can be specified in the devfile's component container; then we would need to update 1. image stream name 2. build config output image stream tag 3. deployment's container image in `pkg/server/devfile-handler.go` and also the annotation mapping for the `ImageStreamTag` in `getTriggerAnnotation()` in `frontend/packages/dev-console/src/components/import/import-submit-utils.ts`. This allows us to use the image name from devfile.yaml rather than always using the application name.
+
+The above proposed devfile container component can also mentions an endpoint instead of hardcoding a default 8080. This can be updated in `createOrUpdateResources()` in `frontend/packages/dev-console/src/components/import/import-submit-utils.ts`
+
+These devfile information would be parsed and returned by the library and thus ensuring a consistent UX.
+
+### Note:
+The above devfile proposal and POC PR assumes the `BuildConfig` outputs the build to an `ImageStreamTag` which is used by the OpenShift internal registry. To push the image to a non-OpenShift Image Registry, the `BuiidConfig` output can pushed to a private registry. OpenShift [doc](https://docs.openshift.com/container-platform/4.6/builds/managing-build-output.html) outlines how this can be achieved via a `BuildConfig` and pushes the build to a private or dockerhub registry. Pushing to a private registry requires secret configuration from the Docker config, and this OpenShift [doc](https://docs.openshift.com/container-platform/3.11/dev_guide/builds/build_inputs.html#using-docker-credentials-for-private-registries) explains how to achieve it.
```

Comment:
To scope the work on the first phase, we should probably clarify that we are going to implement the ImageStream path only. We are not going to implement the private or dockerhub registry on the first phase of the implementation.

Reply:
Updated